server/utils/Reddit.js

```diff
@@ -0,0 +1,56 @@
+const ENDPOINT = "http://www.reddit.com/r/all/search.json";
+
+// const Reddit = {
+//   search(term, limit) {
+//     fetch(`${ENDPOINT}?q=${term}&limit=${limit}`)
+//       .then((res) => {
+//         if (res.ok) return res.json();
+//         throw Error("Request failed");
+//       })
+//       .then((res) => {
+//         console.log(res.data);
+//         return res.data.children.map((post) => ({
+//           content: post.data.url,
+//           title: post.data.title,
+//           platform: "reddit",
+//           image: post.data.thumbnail, // "self" or "default" if no thumbnail link
+//           date: new Date(post.data.created * 1000),
+//           popularity: post.data.ups,
+//         }));
+//       })
+//       .catch((err) => {
+//         console.log(err.message);
+//       });
+//   },
+// };
+
+const searchRecursive = (term, after, posts) => {
```

Comment:
It would be a good idea to wrap this entire function in a promise (async) so that when it's called you can wait for it to complete.